File: Maximizing XOR.py
'''
Given two integers,  and , find the maximal value of  xor , where  and  satisfy the following condition:
L<=A<=B<=R

Input Format

The input contains two lines;  is present in the first line and  in the second line.

'''

#!/bin/python3

import sys

def maximizingXor(l, r):
    res=0
    for i in range(l, r+1):
        for j in range(i+1, r+1):
            x= i ^ j
            if x > res:
                res=x
    return res
if __name__ == "__main__":
    l = int(input().strip())
    r = int(input().strip())
    result = maximizingXor(l, r)
    print(result)

File: Maximizing XOR.c
/*
Given two integers,  and , find the maximal value of  xor , where  and  satisfy the following condition:

L<=A<=B<=R

*/
#include <stdio.h>

int maximizingXor(int l, int r) {
    int ans=0, x;
    for(int i=l; i<=r; ++i){
        for(int j=i+1; j<=r; ++j){
            x=i^j;
            if(ans<x){
                ans=x;
            }
        }
    }
    return ans;
}

int main() {
    int l; 
    scanf("%i", &l);
    int r; 
    scanf("%i", &r);
    int result = maximizingXor(l, r);
    printf("%d\n", result);
    return 0;
}
